### PR TITLE
feat: DID Document structure

### LINKED ISSUE #3: Define DID Document Structure

### PR DESCRIPTION
**Pull request by rolsonquadras**
Added a minimal DID Document Structure to support agent framework and DID Communtion components

Closes #3

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/diddoc/diddoc.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// New creates a new DID Document with ID supplied DID
+func New(id string) *DIDDoc {
+	return &DIDDoc{
+		Context: "https://w3id.org/did/v1",
+		ID:      id,
+	}
+}
+
+// WithPublicKey append public key to the DID Document
+func (doc *DIDDoc) WithPublicKey(publicKey PublicKey) {
```
Comment (sandrask) [inline pkg/diddoc/diddoc.go]: Are we going to validate mandatory fields before adding public key?

File: pkg/diddoc/diddoc.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	ID        string              `json:"id,omitempty"`
+	PublicKey []PublicKey         `json:"publicKey,omitempty"`
+	Service   []ServiceDescriptor `json:"service,omitempty"`
+}
+
+// PublicKey Definition for Public Keys associated with DID Document
+type PublicKey struct {
+	ID     string `json:"id,omitempty"`
+	Type   string `json:"type,omitempty"`
+	Base58 string `json:"publicKeyBase58,omitempty"`
```
Comment (sandrask) [inline pkg/diddoc/diddoc.go]: Are we only going to support Base58?

Comment (rolsonquadras) [reply pkg/diddoc/diddoc.go]: At this iteration, we only need Base58. Hence added minimal structure.

File: pkg/diddoc/diddoc.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/diddoc/diddoc.go]: It's tricky thing to add Validation here as Users can directly add value using the struct, without using this mem func. At this moment, assuming that the Keys passed has mandatory vals.

**Pull request merged**